### PR TITLE
adding-reversing-linkedlist

### PR DESCRIPTION
**Pull request by Adrenalinerush07**
I have coded a 100% working c++ code with comments for reversing a LinkedList.
Please merge it.
This is my first contribution to an open-source repo yay :)

### CODE REVIEW

**Comment by ming-tsai** [comment]
Hi @akrish4 or @Arsenic-ATG,
Could you help me to check this PR?

**Comment by Adrenalinerush07** [comment]
I have sent the PR please review it.

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    push(&head, 10);
+    reverseList(head);
+}
+
+int32_t main()
+{
+    // ios_base::sync_with_stdio(false);cin.tie(NULL);
+    int tt;
+    tt = 1;
+    cin >> tt;
```
Comment (Arsenic-ATG) [inline linked-lists/c-or-cpp/reverse-linkedlist.cpp:65]: why would someone want to execute same set of instructions multiple times for same test cases ?

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    return current;
+}
+
+void solve()
+{
+    Node *head = NULL;
+    push(&head, 5);
+    push(&head, 7);
+    push(&head, 10);
+    reverseList(head);
```
Comment (Arsenic-ATG) [inline linked-lists/c-or-cpp/reverse-linkedlist.cpp:57]: function `reverseList()` returns a value which is never used making the function useless. 
assign the value to `head` for it to function properly.

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
```
Comment (Arsenic-ATG) [inline linked-lists/c-or-cpp/reverse-linkedlist.cpp:58]: No method present to print the result somewhere to show if the function actually did something or not. 

fix : - 
either create a different function to print the list before and after reversal to show the visual representation of working of the function or just add some likes of code doing the same right here

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
### Issues : - 
1. [Index](https://github.com/MakeContributions/DSA/blob/main/linked-lists/README.md) is not updated as told in [contribution guidlines](https://github.com/MakeContributions/DSA#contribution-guidelines)

2. reversing function works but not using the return value makes it practically useless in this particular program
3. overall program have no visual indication of the changes that are taking place 

(see review comments for more)

if you struggle with any part then feel free to ask for help or more information about something you didn't understood🙂

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp (hunk 3 above)
Comment (Adrenalinerush07) [reply linked-lists/c-or-cpp/reverse-linkedlist.cpp]: I have added a function to print LinkedList before and after running the reversing function, please check :)

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp (hunk 2 above)
Comment (Adrenalinerush07) [reply linked-lists/c-or-cpp/reverse-linkedlist.cpp]: I have committed new changes in which I am storing the new head.

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp (hunk 1 above)
Comment (Adrenalinerush07) [reply linked-lists/c-or-cpp/reverse-linkedlist.cpp]: I have removed this.

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+int32_t main()
+{
+    Node *head = NULL;
+    push(&head, 5);
+    push(&head, 7);
+    push(&head, 10);
+
+    cout << "Linked list before reversing ->" << endl;
+    Node *newhead = reverseList(head);
```
Comment (Arsenic-ATG) [inline linked-lists/c-or-cpp/reverse-linkedlist.cpp:69]: change it to 
```cpp
     cout << "Linked list before reversing ->" << endl;
     printlist(head);
     head = reverseList(head);

     cout << "Linked list after reversing ->" << endl;
     printlist(head);
```

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]

File: linked-lists/c-or-cpp/reverse-linkedlist.cpp (hunk 4 above)
Comment (Adrenalinerush07) [reply linked-lists/c-or-cpp/reverse-linkedlist.cpp]: done

Comment (Arsenic-ATG) [reply linked-lists/c-or-cpp/reverse-linkedlist.cpp]: you missed `printlist()` before reversing

Comment (Adrenalinerush07) [reply linked-lists/c-or-cpp/reverse-linkedlist.cpp]: added printlist()

**Review by Arsenic-ATG — APPROVED** [review]

**Comment by ming-tsai** [comment]
Hi @Adrenalinerush07,
Nice work, thanks for your contributions, and welcome to our team.
> You could see the invitation on this [link](https://github.com/MakeContributions)

**Pull request merged**